### DISCUSSION TITLE
readline.read_history_file(historyPath) OSError: [Errno 22] Invalid argument

### DISCUSSION THREAD

**Discussion by Fak3** [pdm-project/pdm#3125] [Q&A]
I'm leaving this note here just for others encountering same error. Workaround found may not be ideal solution, so  other answers welcome.

I was playing with switching python versions for my project between 3.11 and 3.12. At some point python3.11 installed by pdm started to throw OSError on start:
 
```
$ python
Python 3.11.9 (main, Aug 14 2024, 05:07:28) [Clang 18.1.8 ] on linux
Type "help", "copyright", "credits" or "license" for more information.
Traceback (most recent call last):
  File "/etc/pythonstart", line 30, in <module>
    readline.read_history_file(historyPath)
OSError: [Errno 22] Invalid argument
>>>
```

As a workaround, I removed  `/home/u1/.pyhistory3.11`  - it seems to have resolved the issue.

**Comment by frostming** [comment]
Closing for not related to PDM
